### DISCUSSION TITLE
ERROR - Invalid object ID for a chat

### DISCUSSION THREAD

**Discussion by Chappie18** [aahnik/tgcf#49] [Q&A] [answered]
```shell
2021-04-05 21:04:10,382 - root - ERROR - Invalid object ID for a chat. Make sure to pass the right types, for instance making sure that the request is designed for chats (not channels/megagroups) or otherwise look for a different one more suited\nAn example working with a megagroup and AddChatUserRequest, it will fail because megagroups are channels. Use InviteToChannelRequest instead (caused by SendMessageRequest)
Traceback (most recent call last):
  File "/home/kali/Downloads/tgcf-main/forwarder.py", line 53, in forward_job
    await client.send_message(intify(to_chat), replace(message))
  File "/home/kali/.local/lib/python3.9/site-packages/telethon/client/messages.py", line 804, in send_message
    result = await self(request)
  File "/home/kali/.local/lib/python3.9/site-packages/telethon/client/users.py", line 30, in __call__
    return await self._call(self._sender, request, ordered=ordered)
  File "/home/kali/.local/lib/python3.9/site-packages/telethon/client/users.py", line 79, in _call
```
  
Getting this error while copying all messages of a joined private channel to my channel and group

**Comment by aahnik** [comment, marked as answer]
I think you have used the wrong chat ids for the chats.

In your `config.ini`, you should use correct chat ids for the `from` and `to`.

Take help of the [@userinfobot](https://telegram.me/userinfobot) on telegram.

**Reply by Chappie18** [reply]
I have a doubt that, does it work with copying all messages of private channels that I am currently joined to a self created channel.

**Reply by aahnik** [reply]
it works for both private and public channels, as long as you are a member of the channel.

**Reply by lehuuhai014** [reply]
Hi @aahnik , thank you! such a great project.
The forward function work so smoothly. However, I want to ask the setting of config.yml when forwards from the private channel that we and the telegram bot aren't the admin of that channel?

**Reply by lehuuhai014** [reply]
forwards:
  - source: -1001232564717
    dest: [-1001433758953]

It works when I'm the admin of the 2 channels above and put the bot telegram into those two channels.
If one of the two-channel I don't put bot telegram into it, then the forward function will not work.

**Reply by ankit-sinha-18** [reply]
Use session string in that case

**Reply by lehuuhai014** [reply]
Yeah, I already put the SESSION_STRING value into the .env file.

**Reply by ankit-sinha-18** [reply]
Then don'use bot token

**Reply by lehuuhai014** [reply]
My .env filed file:
API_ID
API_HASH
SESSION_STRING

In my case, The bot telegram has to be an admin of the source and dest channel too. If i remove one of them, it will not work.

**Reply by aahnik** [reply]
For the source channel, you don't need to be admin. You just need access to the channel.

For the destination channel, you must have write access.

tgcf also works for groups, or normal chats.

**Reply by lehuuhai014** [reply]
Thank you a lot!

**Reply by aahnik** [reply]
hi @lehuuhai014 , please read the wiki! and more details will be added by this month. video is also coming. you may star and watch tgcf on github, to get notified of updates.

**Reply by lehuuhai014** [reply]
Yes!
